packages/coreutils/src/nbformat.ts

```diff
@@ -24,17 +24,17 @@ export namespace nbformat {
   /**
    * The kernelspec metadata.
    */
-  export interface IKernelspecMetadata extends JSONObject {
+  export interface IKernelspecMetadata extends PartialJSONObject {
```

Comment:
Should this be `PartialJSONObject`, *i.e.*, can its attributes be `undefined`?

Reply:
I guess another way of asking this is: should the attributes be optional, instead?

Reply:
As far as I can tell from the docs (https://jupyter-client.readthedocs.io/en/stable/kernels.html#kernelspecs), `name` and `display_name` are the only required attributes, but there might be others as well, which will then be optional props (requiring partial types). If the docs are inaccurate, or if I am reading them wrong, I am happy to tighten the typings.

Reply:
> requiring partial types

Because:
- they require a partial indexing signature
- the substructures might be partial, which would require that this structure is partial

Reply:
Gotcha, thanks for explaining it. I didn't realize other attributes were possible.